lib/streamlit/DeltaGenerator.py

```diff
@@ -1777,6 +1801,140 @@ def text_input(self, element, label, value=""):
+        element,
+        label,
+        value=NoValue(),
+        min_value=None,
+        max_value=None,
+        step=None,
+        format=None
+    ):
+        """Display a single-line numeric input widget.
+
+        Parameters
+        ----------
+        label : str or None
+            A short label explaining to the user what this input is for.
+        value : int/float or None
+            The value of this widget when it first renders.
+            default: min_value or 0
+        min_value : int/float or None
+            The minimum permitted value.
+        max_value : int/float or None
+            The maximum permitted value.
+        step : int/float or None
+            The stepping interval.
+            Defaults to 1 if the value is an int, 0.01 otherwise.
+            If the value is not specified, the format parameter will be used.
+        format : str or None
+            Printf/Python format string.
+
+
+        Returns
+        -------
+        int/float
+            The current value of the numeric input widget.
+
+        Example
+        -------
+        >>> number = st.number_input('Insert a number')
+        >>> st.write('The current number is ', number)
+
+        """
+
+        if isinstance(value, NoValue):
+            if min_value:
+                value = min_value
+            else:
+                value = 0
+
+        value = 0 if isinstance(value, NoValue) else value
+        int_value = isinstance(value, int)
+        float_value = isinstance(value, float)
+
+        if value is None:
+            raise ValueError("The value should either be an int/float")
+        else:
+            if format is None:
+                format = "%d" if int_value else "%0.2f"
+
+            if step is None:
+                if format is None:
+                    step = 1 if int_value else 0.01
+                else:
+                    step = 1 if is_int(format % 0.01) else 0.01
+
+        # Ensure that all arguments are of the same type.
+        args = [min_value, max_value, step]
+
+        int_args = all(map(lambda a: (
+            isinstance(a, int) or isinstance(a, type(None))), args))
+        float_args = all(map(lambda a: (
+            isinstance(a, float) or isinstance(a, type(None))), args))
+
+        if not int_args and not float_args:
+            raise TypeError(
+                "All arguments must be of the same type."
+                "\n`value` has %(value_type)s type."
+                "\n`min_value` has %(min_type)s type."
+                "\n`max_value` has %(max_type)s type."
+                % {
+                    "value_type": type(value).__name__,
+                    "min_type": type(min_value).__name__,
+                    "max_type": type(max_value).__name__,
+                }
+            )
+
+        # Ensure that the value matches arguments' types.
+        all_ints = int_value and int_args
+        all_floats = float_value and float_args
+
+        if int_value or float_value:
```

Comment:
Is it possible for both of these to be false?

Reply:
that's was because the value could be None, but now we are specifying a default, i will fix this